### ISSUE TITLE
Support custom numpydoc section names used by mkdocstrings

### ISSUE THREAD

**Issue by alekpikl** [jsh9/pydoclint#287]
When using mkdocstrings with numpy-style docstrings, non-standard section names like Inputs and Outputs are valid and render as admonitions in the generated docs.
  pydoclint does not recognise them as known sections and instead attempts to parse their contents as a Parameters block, producing false positives.

  Example:
```
  class MyOp:
      """Apply some operation to a tensor.

      Parameters
      ----------
      scale
          Scaling factor applied to the output.

      Inputs
      ------
      x
          Input tensor of shape ``(batch, features)``.

      Outputs
      -------
      y
          Output tensor, same shape as ``x``.

      Raises
      ------
      ValueError
          If ``scale`` is not positive.
      """

      def __init__(self, scale: float) -> None:
          if scale <= 0:
              raise ValueError(f"scale must be positive, got {scale}")
          self.scale = scale
```

  Running pydoclint produces:
```
  25: DOC102: Method `MyOp.__init__`: Docstring contains more arguments than in function signature.
  25: DOC103: Method `MyOp.__init__`: Docstring arguments are different from function arguments. Arguments in the docstring but not in the function signature: [------: ,
  -------: , Inputs: , Outputs: , x: , y: ].
```
  The entries under Inputs/Outputs (x, y) are misidentified as argument names that don't match the __init__ signature. If the Raises section is also absent, DOC501/DOC503
   fire as a secondary effect of the same misparse.

  The only workaround is # noqa: DOC102, DOC103 on the __init__ definition, which is noisy and obscures legitimate future violations.

  Request: Either recognise and skip unknown section names gracefully, or add a config option to declare custom section names that pydoclint should treat as opaque prose
  rather than argument blocks.

  pydoclint version: 0.8.3
  style: numpy

**Comment by jsh9** [comment]
Hi, thanks for reporting this.

In https://github.com/jsh9/pydoclint/pull/290, I updated pydoclint's logic so that the violation message is no longer confusing. With the new version, pydoclint would directly tell you DOC001 (docstring parsing error).

Regarding your request of "either recognise and skip unknown section names gracefully, or add a config option to declare custom section names that pydoclint should treat as opaque prose": I've decided to not implement either one.

Skipping unknown sections doesn't align with PEP 20 ([the Zen of Python](https://peps.python.org/pep-0020/)): "explicit is better than implicit".

And "adding a config option to declare custom section names" is something that I don't want to encourage.  It's a slippery slope.  I'd suggest sticking with standard numpydoc section names.  For your case, documenting the input and output tensors in the "Notes" section seems appropriate.

**Comment by jsh9** [comment]
I'm closing this issue as "won't fix", but please feel free to continue the discussion here, if you have more comments.
